whetstone/compiler/src/main/kotlin/com/freeletics/mad/whetstone/WhetstoneCodeGenerator.kt

```diff
@@ -214,4 +223,25 @@ public class WhetstoneCodeGenerator : CodeGenerator {
     private fun FqName.packageString(): String {
         return pathSegments().joinToString(separator = ".")
     }
+
+    private fun TopLevelFunctionReference.parameterTypes(): List<ComposableParameter> {
+        return parameters
+            .filter { it.name != null && it.name != "state" && it.name != "sendAction" }
+            .map {
+                val fqName = it.typeReference?.fqNameOrNull(module)
+                val packageString = fqName?.packageString()?.substringBeforeLast(".")
+                val shortName = fqName?.shortName()?.asString()
```

Comment:
You could use `asClassNameOrNull` here and then just throw the exception for the null case https://github.com/square/anvil/blob/main/compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/KotlinPoetUtils.kt#L41

Reply:
This one is private. I could use the `asClassName` which is above it. However it throws it's own exception

Reply:
I am using `asClassName` now. If `fqName` is null we throw our exception. If it is not null but the ClassName can't be found then `asClassName` throw its exception

Reply:
👍